### PR TITLE
Refactor/t-tooltip

### PR DESCRIPTION
**Pull request by gau94rav**
Remove un-necessary code and clean methods.

### CODE REVIEW

File: library/components/TTooltip.vue [hunk 1]
```diff
@@ -38,17 +37,15 @@ export default {
 			type: String,
 			default: "top"
 		},
+		open: { // To control state from a component.
```
Comment (egcastro) [inline library/components/TTooltip.vue:40]: I'd like to use `isOpen` for boolean states. Summarised how we can named boolean props:

Use prefix is, can, has:

-  is: for visual/behavior variations. e.g. isVisible, isEnable, isActive
- can: fore behavior variations or conditional visual variations. e.g. canToggle, canExpand, canHaveCancelButton
- has: for toggling UI elements. e.g. hasCancelButton, hasHeader

File: library/components/TTooltip.vue [hunk 2]
```diff
@@ -83,27 +80,36 @@ export default {
...
+			this.isVisible = true;
 		},
 		hide() {
-			this.showTooltip = false;
-            this.isFocused = false;
+			this.isVisible = false;
+		}
+	},
+	watch: {
+		open(isOpen) {
```
Comment (egcastro) [inline library/components/TTooltip.vue:108]: I couldn't see from where this `isOpen` comes from

Comment (gau94rav) [reply library/components/TTooltip.vue:108]: You the parameter? the prop `open` is being watched and it's value `isOpen` can be true or false.

File: library/components/TTooltip.vue (hunk 1 above)
Comment (gau94rav) [reply library/components/TTooltip.vue:40]: Yes, probably best to use that prefix because it is just like our other components, @sareg0 agree?

Comment (sareg0) [reply library/components/TTooltip.vue:40]: sounds good

File: library/components/TTooltip.vue [hunk 3]
```diff
@@ -83,27 +80,36 @@ export default {
...
+			this.isVisible = false;
+		}
+	},
+	watch: {
+		isOpen(open) {
+			if (open) {
+				return this.show();
+			}
+			return this.hide();
 		}
```
Comment (sareg0) [inline library/components/TTooltip.vue:113]: I don't think you need this watcher

File: library/components/TTooltip.vue [hunk 4]
```diff
@@ -38,17 +37,15 @@ export default {
...
+		isOpen: { // To control state from a component.
+			type: Boolean,
+			default: false,
+		}
 	},
 	data: () => ({
 		positions: {},
-		showTooltip: false,
-        isFocused: false,
+		isVisible: false,
```
Comment (sareg0) [inline library/components/TTooltip.vue:47]: Can you do this instead and skip the watcher?
```suggestion
		isVisible: isOpen,
```

Comment (gau94rav) [reply library/components/TTooltip.vue:47]: If we use this method, then placement logic might not work and depending on window size and position of elements at that point, tooltip will not be able to align itself properly.

I am sure placement logic can be handled better, though it requires a lot more time I think to change.

Comment (gau94rav) [reply library/components/TTooltip.vue:47]: Watcher makes sure that placement is also triggered on change.

Comment (gau94rav) [reply library/components/TTooltip.vue:47]: Oh wait, sorry I forgot that I changed that, you are right, trying this approach.

**Review by sareg0 — APPROVED** [review]

**Pull request merged**